### PR TITLE
Aliases in services, first unit tests for builtins

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: particle-services/src/app_services.rs [hunk 1]
```diff
@@ -316,9 +319,135 @@ fn get_service_interface(
         blueprint_id: &service.blueprint_id,
         service_id,
         owner_id: &service.owner_id,
+        aliases: &service.aliases,
```
Comment (folex) [inline particle-services/src/app_services.rs:322]: I don't think this is needed, really. It's not a part of the interface in any way. I'm not sure that owner_id & blueprint_id are needed, too.

File: particle-services/src/app_services.rs [hunk 2]
```diff
@@ -316,9 +319,135 @@ fn get_service_interface(
...
+
+    fn create_pid() -> PeerId {
+        let keypair = Keypair::generate_ed25519();
+        let peer_id = PeerId::from(keypair.public());
+        peer_id
+    }
+
+    fn create_pas(local_pid: PeerId, management_pid: PeerId) -> ParticleAppServices {
+        let dir = TempDir::new("test").unwrap();
+        let dir2 = TempDir::new("test").unwrap();
```
Comment (folex) [inline particle-services/src/app_services.rs:353]: could you give meaningful names here? plz

File: particle-services/src/app_services.rs [hunk 3]
```diff
@@ -316,9 +319,135 @@ fn get_service_interface(
...
+        }
+    }
+
+    fn call_add_alias_raw(is_client_manager: bool, args: Vec<JValue>) -> RetStruct {
+        let local_pid = create_pid();
+        let management_pid = create_pid();
+        let pas = create_pas(local_pid, management_pid);
+
+        let client_pid;
+        if is_client_manager {
```
Comment (folex) [inline particle-services/src/app_services.rs:408]:
```suggestion
    fn call_add_alias_raw(as_manager: bool, args: Vec<JValue>) -> RetStruct {
        let local_pid = create_pid();
        let management_pid = create_pid();
        let pas = create_pas(local_pid, management_pid);

        let client_pid;
        if as_manager {
```

File: particle-services/src/app_services.rs [hunk 4]
```diff
@@ -316,9 +319,135 @@ fn get_service_interface(
...
+        ]);
+        assert_eq!(resp.ret_code, 1);
+        assert!(
+            resp.error.contains("Service with id") && resp.error.contains("not found"),
+            "Closure should not found a service to add alias `{}`",
+            resp.error
+        );
+    }
+
+    // TODO add more tests
```
Comment (folex) [inline particle-services/src/app_services.rs:452]: Could you elaborate here? Like give a list of possible tests:
```
// TODO: add more tests
//       - create_service success & fail
//       - get_modules success & fail?
//       - ...
```

If not, maybe remove this TODO?

**Review by justprosh — APPROVED** [review]

**Pull request merged**